src/device.rs

```diff
@@ -21,6 +22,22 @@ pub struct Device {
     device: *mut ffi::udev_device,
 }
 
+/// Permissible types of UNIX file I/O API device special file.
+///
+/// Implements TryFrom, allowing construction with e.g.
+/// ```
+/// let d :DeviceType = b'c'.try_from()?;
+/// ```
+/// See also [`from_devnum`][crate::Device::from_devnum].
+#[derive(TryFromPrimitive)]
```

Comment:
I think the additional dependency is not really worth this single derive. The enum is also super small, I don't think it would be huge burden for downstream to write a two-line `match`-statement, if anyone needs this.